packages/dara-core/dara/core/internal/routing.py

```diff
@@ -170,7 +167,17 @@ async def get_auth_components():  # pylint: disable=unused-variable
         return config.auth_config.component_config
 
     @core_api_router.get('/components', dependencies=[Depends(verify_session)])
-    async def get_components():  # pylint: disable=unused-variable
+    async def get_components(name: Optional[str]=None):  # pylint: disable=unused-variable
+        """
+        If name is passed, will only return the specific component
+
+        :param name: the name of component
+        """
+        if name is not None:
+            registry_mgr: RegistryLookup = utils_registry.get('RegistryLookup')
+            comp = await registry_mgr.get(component_registry, name)
+            return {'name': comp.dict(exclude={'func'})}
```

Comment:
Actually if you returned the entire registry here as well, you could skip the second refetch on the frontend. You're not using the returned single component anyway? 
Though make sure to handle the component being missing in the registry, we probably want to return the entire registry anyway no matter what.

```tsx
// first time try asking directly for the missing component
if (i === 0) {
    
    const res = await request(`/api/core/components?name=${instance.name}`, { method: HTTP_METHOD.GET }, token);
    registry = await res.json();
} else {
  // otherwise just keep retrying the entire registry - existing code
  
}
i++;
```